src/phoenix/experimental/evals/evaluators.py

```diff
@@ -0,0 +1,172 @@
+import json
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Optional, Protocol, cast
+
+from .models.openai import OpenAIModel, OpenAIResponse
+from .templates import PromptTemplate
+
+Record = Mapping[str, Any]
+
+
+@dataclass
+class Eval:
+    """
+    An evaluation and optional explanation. Examples may include relevance,
+    hallucination, or toxicity classifications.
+    """
+
+    value: str
+    explanation: Optional[str] = None
+
+
+class Evaluator(Protocol):
+    """
+    An interface for evaluators, which evaluate individual records, i.e., data
+    points.
+    """
+
+    def evaluate(self, record: Record) -> Eval:
+        ...
+
+
+class LLMFunctionCallingEvaluator:
+    def __init__(
+        self,
+        model: OpenAIModel,
+        template: PromptTemplate,
+        rails: List[str],
+        function_name: str,
+        function_description: str,
+        argument_name: str,
+        argument_description: str,
```

Comment:
Feels like it would be nice to group these in a function definition and we could add helpers like this: https://github.com/run-llama/llama_index/blob/91649e1d7f0a97461ede59191e7da1413672c242/llama_index/tools/function_tool.py#L42